### DISCUSSION TITLE
关于正文中参考文献编号压缩格式的问题

### DISCUSSION THREAD

**Discussion by Level5MM** [mengchaoheng/SCUT_thesis#56] [General]
在论文正文中，如果在同一处插入两篇编号相邻的文献，如[1]和[2]，按照规范应该是显示为[1, 2]才对，模板中默认生成的是[1-2]，尝试试过命令\biboptions{numbers,sort&compress}进行处理，但是编译失败了，有大佬知道这个问题要怎么处理吗？

**Comment by mengchaoheng** [comment]
> 如果在同一处插入两篇编号相邻的文献，如[1]和[2]，按照规范应该是显示为[1, 2]才对

规范：
（1）标注格式：引用参考文献标注方式应全文统一，标注的格式为[序号]，放在引文或转述观点的最后一个句号之前，所引文献序号用小4号Times New Roman体、以上角标形式置于方括号中，如：“…成果[1]”。
（2）序号：标注的序号可选择如下两种方式之一：根据在正文中被首次引用出现的先后次序递增，或者按第一作者姓的英文字母或拼音字母的英文字母顺序递增。如一篇论著在论文中多处引用时，以第一次出现的序号为准；当某一论述同时引证多篇参考文献时，应采用[1， 2，…]的格式，标注方括号中的序号按增序排列。
（3）当提及的参考文献为文中直接说明时，其序号应该与正文排齐，如“由文献[8, 10-14]可知：”。

您如何理解（2）和（3）？
@Level5MM

**Comment by mengchaoheng** [comment]
我的理解是：通常，连续时应该是-，不连续多篇用逗号隔开，这是平时看论文的经验。规范（2）意指按增序排列逗号隔开，（3）意指多篇连续可以用杠-。

**Reply by Level5MM** [reply]
规划（2）增序没错，但我见到的对于仅有两个连续编号，用逗号隔开，有三个以上连续编号，则用杠表示范围。我还尝试用biblatex包里面的一些选项，能变成我上面说的这种样子，但是就没有上标格式了。不知大佬知不知道相关的解决方法？没有的话我晚点再研究研究，谢谢了

**Reply by sikouhjw** [reply]
> 但我见到的对于仅有两个连续编号，用逗号隔开，有三个以上连续编号，则用杠表示范围。

是学校的规定？还是某些论文长这样？某些样本可以不代表必须这样

**Reply by Level5MM** [reply]
看GBT7714-2015确认了一下，里面仅有两个连续编号的例子给的是用杠的，不过里面有一句“此规则不适用于用计算机自动编码的序号”，那我还是继续用模板默认的格式，谢谢大佬

**Comment by mengchaoheng** [comment]
> 但我见到的对于仅有两个连续编号，用逗号隔开，有三个以上连续编号，则用杠表示范围。

您也没贴出来在哪见到的。也许您更喜欢这种引用风格，无可厚非，只要没规定也没禁止的风格都可以采用。由于这不是我所理解的规范，所以我就不探讨这种格式的代码了。

**Reply by Level5MM** [reply]
谢谢大神解答，我继续使用默认格式
